php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1011 @@
+            b.token(PHPKeyword.TRY),
+            BLOCK(),
+            b.zeroOrMore(CATCH_BLOCK()),
+            b.optional(f.newTuple2(b.token(PHPKeyword.FINALLY), BLOCK()))));
+  }
+
+  public CatchBlockTree CATCH_BLOCK() {
+    return b.<CatchBlockTree>nonterminal(PHPLexicalGrammar.CATCH_BLOCK)
+        .is(f.catchBlock(
+            b.token(PHPKeyword.CATCH),
+            b.token(PHPPunctuator.LPARENTHESIS),
+            NAMESPACE_NAME(),
+            b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+            b.token(PHPPunctuator.RPARENTHESIS),
+            BLOCK()
+        ));
+
+  }
+
+  public BlockTree BLOCK() {
+    return b.<BlockTree>nonterminal(PHPLexicalGrammar.BLOCK)
+        .is(f.block(
+            b.token(PHPPunctuator.LCURLYBRACE),
+            b.zeroOrMore(INNER_STATEMENT()),
+            b.token(PHPPunctuator.RCURLYBRACE)
+        ));
+  }
+
+  public GotoStatementTree GOTO_STATEMENT() {
+    return b.<GotoStatementTree>nonterminal(PHPLexicalGrammar.GOTO_STATEMENT)
+        .is(f.gotoStatement(b.token(PHPKeyword.GOTO), b.token(PHPLexicalGrammar.IDENTIFIER), EOS()));
+  }
+
+  public ExpressionStatementTree EXPRESSION_STATEMENT() {
+    return b.<ExpressionStatementTree>nonterminal(PHPLexicalGrammar.EXPRESSION_STATEMENT)
+        .is(f.expressionStatement(EXPRESSION(), EOS()));
+  }
+
+  public LabelTree LABEL() {
+    return b.<LabelTree>nonterminal(PHPLexicalGrammar.LABEL)
+        .is(f.label(b.token(PHPLexicalGrammar.IDENTIFIER), b.token(PHPPunctuator.COLON)));
+  }
+
+  public InternalSyntaxToken EOS() {
+    return b.<InternalSyntaxToken>nonterminal(PHPLexicalGrammar.EOS)
+        .is(b.firstOf(b.token(PHPPunctuator.SEMICOLON), b.token(PHPLexicalGrammar.INLINE_HTML)));
+  }
+
+  /**
+   * [ END ] Statement
+   */
+
+  public ExpressionTree EXPRESSION() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.EXPRESSION)
+      .is(f.expression(b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER)));
+  }
+
+  /**
+   * [ START ] Expression
+   */
+
+  public ExpressionTree COMMON_SCALAR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.COMMON_SCALAR)
+      .is(b.firstOf(
+          f.heredocLiteral(b.token(PHPLexicalGrammar.HEREDOC)),
+          NUMERIC_LITERAL(),
+          STRING_LITERAL(),
+          f.booleanLiteral(b.token(PHPLexicalGrammar.BOOLEAN_LITERAL)),
+          f.nullLiteral(b.token(PHPLexicalGrammar.NULL)),
+          f.magicConstantLiteral(b.firstOf(
+              b.token(PHPLexicalGrammar.CLASS_CONSTANT),
+              b.token(PHPLexicalGrammar.FILE_CONSTANT),
+              b.token(PHPLexicalGrammar.DIR_CONSTANT),
+              b.token(PHPLexicalGrammar.FUNCTION_CONSTANT),
+              b.token(PHPLexicalGrammar.LINE_CONSTANT),
+              b.token(PHPLexicalGrammar.METHOD_CONSTANT),
+              b.token(PHPLexicalGrammar.NAMESPACE_CONSTANT),
+              b.token(PHPLexicalGrammar.TRAIT_CONSTANT)))));
+  }
+
+  public LiteralTree NUMERIC_LITERAL() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)